Source/EasyNetQ/Internals/TaskHelpers.cs

```diff
@@ -96,54 +125,64 @@ public static class TaskHelpers
                 {
                     tcs.SetException(exception);
                 }
+
                 return tcs.Task;
             };
         }
 
-        public static Task FromCancelled()
+        /// <summary>
+        ///     This is an internal API that supports the EasyNetQ infrastructure and not subject to
+        ///     the same compatibility as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new EasyNetQ release.
+        /// </summary>
+        public static void AttachCancellation<T>(
+            this TaskCompletionSource<T> taskCompletionSource, CancellationToken cancellationToken
+        )
         {
-            var tcs = new TaskCompletionSource<object>();
-            tcs.SetCanceled();
-            return tcs.Task;
-        }
+            if (
+                (taskCompletionSource.Task.CreationOptions & TaskCreationOptions.RunContinuationsAsynchronously) !=
+                TaskCreationOptions.RunContinuationsAsynchronously
```

Comment:
```suggestion
                taskCompletionSource.Task.CreationOptions.HasFlag(TaskCreationOptions.RunContinuationsAsynchronously)
```

Reply:
It will box ‘TaskCreationOptions’, so it’s preferred way to check.

Reply:
No more. It was optimized a long time ago https://github.com/dotnet/coreclr/pull/13748

Reply:
Done